syntax = "proto3";

package uoservice;

message Empty {}

message GrpcMobileData {
  string name = 1;
  uint32 x = 2;
  uint32 y = 3;
  uint32 race = 4;
}

message GrpcMobileList {
  repeated GrpcMobileData mobile = 1;
}

message ImageRequest {
  string name = 1;
}

message ScreenImage {
  bytes image = 1;
}

message States {
  ScreenImage screenImage = 1;
  GrpcMobileList mobileList = 2;
}

message MousePoint {
  uint32 x = 1;
  uint32 y = 2;
}

message Actions {
  uint32 action = 1;
  MousePoint mousePoint = 2;
}

service UoService {
  rpc reset (ImageRequest) returns (States);
  rpc step (ImageRequest) returns (States);
  rpc act (Actions) returns (Empty);
}